src/main/java/net/nextencia/rrdiagram/Server.java

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) YugaByte, Inc.
+
+package net.nextencia.rrdiagram;
+
+import net.nextencia.rrdiagram.grammar.model.*;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import java.net.InetSocketAddress;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.File;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.Executors;
+import java.util.logging.Logger;
+import java.util.logging.Level;
+import java.util.logging.LogManager;
+import java.util.logging.Handler;
+import java.util.List;
+import java.util.Map;
+import java.util.HashMap;
+import java.net.URLDecoder;
+import java.net.URI;
+
+interface HttpCodes {
+  int OK = 200;
+  int NOT_FOUND = 404;
+}
+
+class Helper {
+  public static boolean writeHTTPResponse(HttpExchange httpExchange, int responseCode, String content, String contentType) {
+    try {
+      OutputStream outputStream = httpExchange.getResponseBody();
+      httpExchange.getResponseHeaders().set("Content-Type", contentType);
+      httpExchange.sendResponseHeaders(responseCode, content.length());
+      outputStream.write(content.getBytes());
+      outputStream.flush();
+      outputStream.close();
+      return true;
+    } catch(java.io.IOException e) {
+      e.printStackTrace(System.out);
+      return false;
+    }
+  }
+
+  public static boolean writeHTTPResponse(HttpExchange httpExchange, int responseCode, String content) {
+    return writeHTTPResponse(httpExchange, responseCode, content, "text/plain; charset=us-ascii");
+  }
+}
+
+class BNFHandler implements HttpHandler {
+
+  Logger logger = Logger.getLogger(BNFHandler.class.getName());
+
+  @Override
+  public void handle(HttpExchange httpExchange) throws IOException {
+    if("GET".equals(httpExchange.getRequestMethod())) { 
+      handleRequest(httpExchange);
+    }
+  }
+
+  public Map<String, String> parseParameters(HttpExchange exchange) {
+    URI requestedUri = exchange.getRequestURI();
+    String query = requestedUri.getRawQuery();
+    Map<String, String> parameters = new HashMap<String, String>();
+     if (query != null) {
+       String pairs[] = query.split("[&]");
+
+       for (String pair : pairs) {
+        String param[] = pair.split("[=]");
```

Comment:
```suggestion
        String param[] = pair.split("=");
```

read the documentation.  probably above one can be `"&"` as well

Reply:
url is `?k1=v1&k2=v2` , `&` extracts kv pairs and `=` splits kv-pairs into key & value.